agnpy/emission_regions/blob.py

```diff
@@ -393,3 +397,50 @@ def u_ph_synch(self):
             * np.trapz(np.power(self.gamma_e, 2) * self.n_e(self.gamma_e), self.gamma_e)
         )
         return u_ph.to("erg cm-3")
+
+    def n_e_time_evolution(self, energy_loss_function, time, subintervals_count=10):
+        """Performs the time evolution of the electron distribution inside the blob.
+
+        Parameters
+        ----------
+        energy_loss_function : function
```

Comment:
this part you might need to rethink. You specify here a single function, while in reality there will be different processes to be considered and they will have different effect at different moments and at varying energies. So either specify here a list of functions (I guess it is possible??), and loop over them, or a user will need to write a function where he adds a few processes and specifies this function here.

Reply:
Yes I agree, this will need to be changed, but for now it's easier to use it and test it this way, and it's also quite generic. But when we add support for more processes, this part will need some rethinking, how to combine such multiple processes.